clang/lib/AST/Expr.cpp

```diff
@@ -1639,11 +1639,19 @@ SourceLocation CallExpr::getBeginLoc() const {
   if (const auto *OCE = dyn_cast<CXXOperatorCallExpr>(this))
     return OCE->getBeginLoc();
 
+  if (const auto *Method =
+          dyn_cast_if_present<const CXXMethodDecl>(getCalleeDecl());
```

Comment:
nit: I presume the `const` in `<const CXXMethodDecl>` isn't necessary, right?

Reply:
Maybe you could do something like

```
unsigned Offset = 0;
if(isExplicitObjectMemberFunction())
    Offset = 1;
```
To avoid some duplication

Reply:
Sorry, I'm not quite following this suggestion. What am I supposed to do with `Offset`?

Reply:
heh I think there is some opportunities to merge the slightly different if statements (for the explicit and not explicit cases) but it would be a minor improvement so I'll will approve that

Reply:
> heh I think there is some opportunities to merge the slightly different if statements (for the explicit and not explicit cases) but it would be a minor improvement so I'll will approve that

Ah, I see.

I considered doing something like this:

```c++
bool UseFirstArgLoc = false;
if (/* is explicit object */)
  UseFirstArgLoc = true;

SourceLocation begin = getCallee()->getBeginLoc();
if ((begin.isInvalid() || UseFirstArgLoc) && getNumArgs() > 0 && getArg(0))
  begin = getArg(0)->getBeginLoc();
return begin;
```
  
But this needlessly evaluates `getCallee()->getBeginLoc()` in the explicit case, so I opted not to do this.